test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go

```diff
@@ -132,66 +133,130 @@ var _ = Describe("Mixedcpus", Ordered, func() {
+
+		When("Modifying the shared CPUs in the performance profile", func() {
+			BeforeEach(func() {
+
+				isolated := mustParse(string(*profile.Spec.CPU.Isolated))
+				shared := mustParse(string(*profile.Spec.CPU.Shared))
+				// select one arbitrary
+				isolatedCore := mustParse(strconv.Itoa(isolated.List()[0]))
+				sharedCore := mustParse(strconv.Itoa(shared.List()[0]))
+				// swap one from the isolated with one from the shared
+				//  to create new shared cpuset
+				newIsolated := isolated.Difference(*isolatedCore).Union(*sharedCore)
+				newShared := shared.Difference(*sharedCore).Union(*isolatedCore)
+				profile.Spec.CPU.Isolated = cpuSetToPerformanceCPUSet(&newIsolated)
+				profile.Spec.CPU.Shared = cpuSetToPerformanceCPUSet(&newShared)
+				By("applying new performanceProfile")
+				testprofiles.UpdateWithRetry(profile)
+				mcp, err := mcps.GetByProfile(profile)
+				Expect(err).ToNot(HaveOccurred())
+				By("waiting for mcp to catch up")
+				mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+				mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+				Expect(testclient.Client.Get(ctx, client.ObjectKeyFromObject(profile), profile))
+				testlog.Infof("new isolated CPU set=%q\nnew shared CPU set=%q", newIsolated.String(), newShared.String())
+				// we do not bother to revert the profile at the end of the test, since its irrelevant which of the cpus are shared
+			})
+
+			It("should contains the updated values under the container", func() {
+				rl := &corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("1"),
+					corev1.ResourceMemory: resource.MustParse("100Mi"),
+					sharedCpusResource:    resource.MustParse("1"),
+				}
+				p, err := createPod(ctx, testclient.Client, testutils.NamespaceTesting,
+					withRequests(rl),
+					withLimits(rl),
+					withRuntime(components.GetComponentName(profile.Name, components.ComponentNamePrefix)))
+				Expect(err).ToNot(HaveOccurred())
+
+				By("checking cgroups under the container")
+				ppShared := mustParse(string(*profile.Spec.CPU.Shared))
+				cfg := &controller.CpuSet{}
+				err = getter.Container(ctx, p, p.Spec.Containers[0].Name, cfg)
+				Expect(err).ToNot(HaveOccurred())
+				cgroupCpuSet := mustParse(cfg.Cpus)
+				Expect(cgroupCpuSet.Intersection(*ppShared).List()).ToNot(BeEmpty(), "shared cpus are not in the pod cgroups; pod=%q, cgroupscpuset=%q sharedcpuset=%q",
+					fmt.Sprintf("%s/%s", p.Namespace, p.Name), cgroupCpuSet.String(), ppShared.String())
+
+				By("checking environment variable under the container")
+				cmd := printMixedCPUsEnvCmd()
+				output, err := pods.ExecCommandOnPod(testclient.K8sClient, p, "", cmd)
+				Expect(err).ToNot(HaveOccurred(), "failed to execute command on pod; cmd=%q pod=%q", cmd, client.ObjectKeyFromObject(p).String())
+				isolatedAndShared := strings.Split(string(output), "\r\n")
+				isolatedFromEnv := mustParse(isolatedAndShared[0])
```

Comment:
same note regarding validating the length.

Reply:
`mustParse` should assert in case of failure

Reply:
right but that is a different error and it hides the detail of the actual root cause